### ISSUE TITLE
Automatic Performance Benchmarking / Testing

### ISSUE THREAD

**Issue by Ljupch0** [DaxStudio/DaxStudio#1107]
Great job on DAX studio, invaluable tool.

tldr: Can I programmatically input a DAX query and obtain server timings?

My setup and idea: I have a report with many pages. I would like to export the Performance Analyzer file once for each page. This file contains the DAX queries that power every visual on the page. I would like to, hopefully programmatically, load this file and get server timings for every visual, containing the overall time and split by FE and SE.

I would then do some optimizations to the measures, and re-run the Performance Analyzer to get the new DAX queries for the page. I would then run the new file and compare improvements in performance in an automated way.

Any ideas how this could be achieved? At the core of my request is to programmatically input a DAX query into a function and obtain server timings data, with clear_cache = TRUE :)

**Comment by marcosqlbi** [comment]
The request makes sense but it requires much more work than it seems.
In general, it would be better to get a command-line version of DAX Studio that can automate the process described.
However, in my experience, I have hardly seen the need for such requirements because a more productive way to optimize DAX measures is to start from the more expensive ones and focus the effort there to maximize productivity.
Please, can you better describe which scenario you would cover? How do you approach the optimization in the use case described? Which measures would you optimize first and based on which analysis?

**Comment by Ljupch0** [comment]
Ciao Marco, thank you for the quick response. I have just started the new edition of Optimizing DAX and I enjoy the content. 

I guess my request is more reporting, ie I could easily document a performance comparison of an optimization of an entire report page. During the optimization, of course I would be manually making changes to the measures, generating the query and checking the performance in DAX Studio with cleared cache. 

In the ideal case I would have such single page performance snapshots at different intervals and easily track the performance over time. Maybe in its own Power BI meta-report.

Do you see a way I can send a DAX query and get server timings, even with some hacking around involved? Thank you!

**Comment by marcosqlbi** [comment]
Thanks for the clear explanation. It makes sense to generate a complete performance assessment on a number of queries.
I don't think there is a way to automate the entire process with DAX Studio. You may try Power Automate Desktop, which could do exactly what you want to do with some hacking :) But even if it works (I'm not sure it will), it will not be ideal.
It is a good idea for a future feature, but as I said, a good implementation requires time and I'm not sure it will get a better prioritization compared to other pending features.

**Comment by Ljupch0** [comment]
Thank you for the exhaustive answer, I will have a look at Power Automate.
